pkg/controller/job/job_controller.go

```diff
@@ -1504,22 +1506,17 @@ func (jm *Controller) manageJob(ctx context.Context, job *batch.Job, jobCtx *syn
 		jm.expectations.ExpectDeletions(logger, jobKey, len(podsToDelete))
 		removed, err := jm.deleteJobPods(ctx, job, jobKey, podsToDelete)
 		active -= removed
-		if feature.DefaultFeatureGate.Enabled(features.JobPodReplacementPolicy) {
+		if trackTerminatingPods(job) {
 			*jobCtx.terminating += removed
 		}
 		return active, metrics.JobSyncActionPodsDeleted, err
 	}
 
 	var terminating int32 = 0
```

Comment:
maybe this can be moved to line 1566 where it's actually needed.

Also, maybe we can just update `wantActive`, as opposed to keeping the additional `terminating` variable?

Reply:
> maybe this can be moved to line 1566 where it's actually needed.

Good point, moved down.

> Also, maybe we can just update wantActive, as opposed to keeping the additional terminating variable?

This would work semantically the same, but I think it could be less readable. It would also mean this log line is changed: https://github.com/kubernetes/kubernetes/blob/78377c4d105533f76d12886f82128469b6f3c0e9/pkg/controller/job/job_controller.go#L1598, which might be confusing.

It there are good idea to make it readable I would like to follow up.